src/hooks/useApplicationHealthStatus.ts

```diff
@@ -1,24 +1,39 @@
-import * as React from 'react';
-import { getGitOpsDeploymentHealthStatusIcon } from '../utils/gitops-utils';
-import { useGitOpsDeploymentCR } from './useGitOpsDeploymentCR';
+import { useMemo } from 'react';
+import { getComponentDeploymentRunStatus } from '../utils/environment-utils';
+import { useNamespace } from '../utils/namespace-context-utils';
+import { useSnapshotsEnvironmentBindings } from './useSnapshotsEnvironmentBindings';
+
+const SNAPSHOT_BINDING_ENV_LABEL = 'appstudio.environment';
 
 export const useApplicationHealthStatus = (
-  namespace: string,
   applicationName: string,
-): [string, React.ReactNode, boolean] => {
-  // TODO: Change from gitOpsDeployment to environment based query
-  const [gitOpsDeployment, gitOpsDeploymentLoaded] = useGitOpsDeploymentCR(
-    applicationName,
-    namespace,
-  );
-
-  const gitOpsDeploymentHealthStatus = gitOpsDeploymentLoaded
-    ? gitOpsDeployment?.status?.health?.status
-    : null;
```

Comment:
In a slack conversation with Jonathan West (https://redhat-internal.slack.com/archives/C02CTEB3MMF/p1675265283752929), he states:
```
HOWEVER, we don't currently provide the health field as part of the SnapshotEnvironmentBinding. For that, you will have to look in the .status.health field of the GitOpsDeployment.
```
so I think we need to still check the GitOpsDeployment setting if the `SnapshotEnvironmentBinding` indicates healthy.

Reply:
added check for gitOps CR as well `.status.health`